footprints/main/models.py

```diff
@@ -994,8 +969,15 @@ def display_title(self):
             return self.title
 
     def owners(self):
-        role = Role.objects.get_owner_role()
-        return self.actor.filter(role=role)
+        return self.actor.filter(
+            role__name=Role.OWNER).select_related('person')
+
+    def owners_list(self):
+        a = ''
+        for actor in self.actor.all():
+            if actor.role.name == Role.OWNER:
+                a += actor.display_name() + ', '
+        return a[:-2]
```

Comment:
It's probably only a small number of actors and we can get away with it, but generally repeated string appends are super inefficient in Python. I think the filtering could also happen at the db. How about:

```
def owners_list(self):
    actors = [actor.display_name() for actor in self.actor.filter(role__name=Role.OWNER)]
    return ', '.join(actors)
```